webui/react/src/components/HyperparameterSearchModal.tsx

```diff
@@ -391,13 +401,33 @@ const HyperparameterSearchModal = ({ closeModal, experiment, trial }: Props): JS
               <h2>Grid Count</h2>
             </label>
           )}
-          {hyperparameters.map((hp) => (
-            <HyperparameterRow key={hp.name} searcher={searcher} {...hp} />
+          <label id="delete" />
+          {currentHPs?.map((hp, idx) => (
+            <HyperparameterRow
+              handleDelete={(name: string) =>
+                setCurrentHPs((prev) => prev?.filter((hp) => hp.name !== name))
+              }
+              key={idx}
+              searcher={searcher}
+              {...hp}
+            />
           ))}
+          <label id="add">
+            <Button
+              onClick={() =>
+                setCurrentHPs((prev) =>
+                  prev
+                    ? [...prev, { hyperparameter: emptyHP, name: `hp_${prev.length}` }]
+                    : [{ hyperparameter: emptyHP, name: 'hp_0' }],
```

Comment:
sugg: you could do `[...(prev ?? []), { hyperparameter: emptyHP, name: `hp_${prev?.length ?? 0}` }]`

Reply:
But that's personal taste, up to you.